micro_sam/sam_3d_wrapper.py

```diff
@@ -6,16 +6,32 @@
 from segment_anything.modeling.image_encoder import window_partition, window_unpartition
 from segment_anything.modeling import Sam
 
-from .util import get_sam_model
-
-
-def get_3d_sam_model(device, n_classes, image_size, model_type="vit_b"):
-    predictor, sam = get_sam_model(
-        return_sam=True, model_type=model_type, device=device, num_multimask_outputs=n_classes,
-        flexible_load_checkpoint=True, image_size=image_size,
+from .util import get_sam_model, _load_checkpoint, _handle_checkpoint_loading
+
+
+def get_3d_sam_model(
+    device,
+    n_classes,
+    image_size,
+    model_type="vit_b",
+    checkpoint_path=None,
+):
+    _, sam = get_sam_model(
+        model_type=model_type,
+        device=device,
+        return_sam=True,
+        flexible_load_checkpoint=True,
+        num_multimask_outputs=n_classes,
+        image_size=image_size,
     )
+
     sam_3d = Sam3DWrapper(sam)
     sam_3d.to(device)
+
+    if checkpoint_path is not None:
```

Comment:
Get rid of this and just load the checkpoint in the beginning?! See comment above.

Reply:
As mentioned above, this won't work unless we pull in the sam3d wrapper inside `get_sam_model`.